File: HackerRank/the_time_in_words.py
#!/bin/python3
# https://www.hackerrank.com/challenges/the-time-in-words/problem

import math
import os
import random
import re
import sys

#
# Complete the 'timeInWords' function below.
#
# The function is expected to return a STRING.
# The function accepts following parameters:
#  1. INTEGER h
#  2. INTEGER m
#

def get_min_string(m):
    ''' Provides a string corresponding to the minute quantity m'''
    if m == 1:
        return ' minute'
    if (m == 15) or (m == 30):
        return ''
    return ' minutes'


def timeInWords(h, m):
    # The words correspond to the list index. Example: list[1] is 'one'
    # Exceptions: list[15] = quarter and list[30] = half
    to_word = ['zero', 'one', 'two', 'three', 'four', 'five', 'six',
    'seven', 'eight', 'nine', 'ten', 'eleven', 'twelve', 'thirteen', 'fourteen',
    'quarter', 'sixteen', 'seventeen', 'eighteen', 'nineteen', 'twenty', 'twenty one',
    'twenty two', 'twenty three', 'twenty four', 'twenty five', 'twenty six',
    'twenty seven', 'twenty eight', 'twenty nine', 'half']

    full_hour = 60  # minutes

    if (m == 0):  # Use o' clock
        return f"{to_word[h]} o' clock"

    if (m <= 30):  # Use past
        compl_string = 'past'
    else:  # Use to
        m = full_hour - m
        h += 1
        compl_string = 'to'

    min_string = get_min_string(m)
    final_time = f'{to_word[m]}{min_string} {compl_string} {to_word[h]}'  # Joins the strings to get the final answer
    return final_time


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    h = int(input().strip())

    m = int(input().strip())

    result = timeInWords(h, m)

    fptr.write(result + '\n')

    fptr.close()

File: HackerRank/The_Time_In_Words.c
https://www.hackerrank.com/challenges/the-time-in-words/problem


#include<stdio.h>


#include<stdlib.h>


#include<string.h>


#include<assert.h>


#include<math.h>


int main()


{


    int h,m;


    scanf("%d\n %d\n",&h,&m);


    char count[30][100]={"one","two","three","four","five","six","seven","eight","nine","ten","eleven","twelve","thirteen","fourteen","fiveteen","sixteen","seventeen","eighteen","nineteen","twenty","twenty one","twenty two","twenty three","twenty four","twenty five","twenty six","twenty seven","twenty eight","twenty nine"};


    if(m==0)


    printf("%s o' clock",count[h-1]);


    else if(m>30)


    {


        if(m==45)


        printf("quarter to %s",count[h]);


        else if(m==59)


        printf("one minute to %s",count[h]);


        else if(m!=45 && m!=59)


        printf("%s minutes to %s",count[60-m-1],count[h]);


    }


    else if(m<=30)


    {


        if(m==1)


        printf("one minute past %s",count[h-1]);


        else if(m==15)


        printf("quarter past %s",count[h-1]);


        else if(m==30)


        printf("half past %s",count[h-1]);


        else if(m!=15 && m!=30 && m!=1)


        printf("%s minutes past %s",count[m-1],count[h-1]);


    }


    return 0;


}

